client/flutter/lib/pages/onboarding/location_sharing_page.dart

```diff
@@ -33,12 +33,16 @@ class _LocationSharingPageState extends State<LocationSharingPage> {
     try {
       await Location().requestPermission();
       if (await Location().hasPermission() == PermissionStatus.granted) {
-        LocationData location = await Location().getLocation();
-        Map jitteredLocationData = JitterLocation().jitter(location.latitude, location.longitude, 5/*kms refers to kilometers*/);
-        
-        await WhoService.putLocation(
-          latitude: jitteredLocationData['lat'],
-          longitude: jitteredLocationData['lng']);
+        if (await Location().requestService()) {
+          LocationData location = await Location().getLocation();
+          Map jitteredLocationData = JitterLocation().jitter(
+              location.latitude, location.longitude,
+              5 /*kms refers to kilometers*/);
+
+          await WhoService.putLocation(
+              latitude: jitteredLocationData['lat'],
+              longitude: jitteredLocationData['lng']);
+        }
       }
     } catch(_) {
       // ignore for now.
```

Comment:
Why is the finally { _complete(); } not enough?

Reply:
> Why is the finally { _complete(); } not enough?

If Location Service is off and the "await Location().getLocation()" is directly called, it will never return thus it will not reach the subsequent code (e.g finally clause).  We should call Location().requestService() first to fix this.

Reply:
Got it!  That def seems like a bug in getLocation to never terminate, but your fix looks good.

Reply:
https://github.com/WorldHealthOrganization/app/blob/master/client/flutter/lib/pages/onboarding/location_sharing_page.dart#L36
The future on this line isn't finishing. 
I tested the change and it all works now, no crash and after location is retrieved the page is closed.